### PR TITLE
Add device id and startup mode on macOS swift app

### PR DESCRIPTION
**Pull request by Cloudwu1**
## Description

Add device id and startup mode on macOS swift app

## Screenshot

Before:
image

After: 
image

### CODE REVIEW

File: SasquatchMac/SasquatchMacSwift/AppDelegate.swift [hunk 1]
```diff
@@ -52,7 +60,25 @@ class AppDelegate: NSObject, NSApplicationDelegate, MSCrashesDelegate, MSPushDel
     }
 
     // Start AppCenter.
-    MSAppCenter.start("7e873482-108f-4609-8ef2-c4cebd7418c0", withServices : [ MSAnalytics.self, MSCrashes.self, MSPush.self ])
+    let appSecret = "7e873482-108f-4609-8ef2-c4cebd7418c0"
```
Comment (MatkovIvan) [inline SasquatchMac/SasquatchMacSwift/AppDelegate.swift]: Move it to the consts file

File: SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift [hunk 2]
```diff
@@ -30,4 +37,18 @@ class AppCenterViewController : NSViewController {
     UserDefaults.standard.set(userId, forKey: "userId")
```
Comment (MatkovIvan) [inline SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift]: Could you please do some cleanup here and use `kMSUserIdKey`?

**Comment by jwallra** [comment]
Please add a screenshot of what you've added to the app

File: SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift [hunk 3]
```diff
@@ -30,4 +37,18 @@ class AppCenterViewController : NSViewController {
...
     appCenter.setUserId(userId)
   }
+  // DeviceID
```
Comment (jwallra) [inline SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift]: New line before comment

File: SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift [hunk 4]
```diff
@@ -30,4 +37,18 @@ class AppCenterViewController : NSViewController {
...
+  class func getDeviceIdentifier() -> String? {
+    let platformExpert = IOServiceGetMatchingService(kIOMasterPortDefault, IOServiceMatching("IOPlatformExpertDevice"))
+    let serialNumberAsCFString = IORegistryEntryCreateCFProperty(platformExpert, kIOPlatformSerialNumberKey as CFString, kCFAllocatorDefault, 0)
+    let baseIdentifier = serialNumberAsCFString?.takeRetainedValue() as! String
+        IOObjectRelease(platformExpert)
+    return baseIdentifier
+  }
+  // Startup Mode
```
Comment (jwallra) [inline SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift]: New line before comment

File: SasquatchMac/SasquatchMac/Base.lproj/SasquatchMac.storyboard [hunk 5]
```diff
@@ -808,10 +808,124 @@
...
+                                <rect key="frame" x="18" y="388" width="64" height="17"/>
+                                <textFieldCell key="cell" scrollable="YES" lineBreakMode="clipping" sendsActionOnEndEditing="YES" title="Device Id:" id="MME-uY-6ba">
+                                    <font key="font" metaFont="system"/>
+                                    <color key="textColor" name="labelColor" catalog="System" colorSpace="catalog"/>
+                                    <color key="backgroundColor" name="controlColor" catalog="System" colorSpace="catalog"/>
+                                </textFieldCell>
+                            </textField>
+                            <textField verticalHuggingPriority="750" translatesAutoresizingMaskIntoConstraints="NO" id="7dF-GM-Y0D">
+                                <rect key="frame" x="100" y="383" width="310" height="22"/>
+                                <textFieldCell key="cell" scrollable="YES" lineBreakMode="clipping" selectable="YES" editable="YES" sendsActionOnEndEditing="YES" state="on" borderStyle="bezel" drawsBackground="YES" id="TdS-HU-Rx6">
```
Comment (jaelim-ms) [inline SasquatchMac/SasquatchMac/Base.lproj/SasquatchMac.storyboard]: editable NO here.

File: SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift (hunk 2 above)
Comment (Cloudwu1) [reply SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift]: Resolve in new commit.

File: SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift (hunk 3 above)
Comment (Cloudwu1) [reply SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift]: Resolve in new commit.

File: SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift (hunk 4 above)
Comment (Cloudwu1) [reply SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift]: Resolve in new commit.

File: SasquatchMac/SasquatchMac/Base.lproj/SasquatchMac.storyboard (hunk 5 above)
Comment (Cloudwu1) [reply SasquatchMac/SasquatchMac/Base.lproj/SasquatchMac.storyboard]: Resolve in new commit.

File: SasquatchMac/SasquatchMacSwift/AppDelegate.swift [hunk 6]
```diff
@@ -5,6 +5,14 @@ import AppCenterAnalytics
 import AppCenterCrashes
 import AppCenterPush
 
+enum StartupMode: Int {
+    case APPCENTER
```
Comment (achocron) [inline SasquatchMac/SasquatchMacSwift/AppDelegate.swift]: Can we use camel case? `appCenter`, `oneCollector`, `both`, `none`, and `skip`?

Comment (Cloudwu1) [reply SasquatchMac/SasquatchMacSwift/AppDelegate.swift]: Resolved in new commit.

**Comment by Cloudwu1** [comment]
conflicts are resolved.

File: SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift [hunk 7]
```diff
@@ -10,13 +10,20 @@ class AppCenterViewController : NSViewController {
...
 
+  @IBOutlet weak var deviceIdField: NSTextField!
+  @IBOutlet weak var startupModeField: NSComboBox!
+
   override func viewDidLoad() {
     super.viewDidLoad()
     installIdLabel?.stringValue = appCenter.installId()
     appSecretLabel?.stringValue = appCenter.appSecret()
     logURLLabel?.stringValue = appCenter.logUrl()
     userIdLabel?.stringValue = UserDefaults.standard.string(forKey: "userId") ?? ""
```
Comment (MatkovIvan) [inline SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift]:
```suggestion
     userIdLabel?.stringValue = UserDefaults.standard.string(forKey: kMSUserIdKey) ?? ""
```

File: SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift [hunk 8]
```diff
@@ -27,7 +34,23 @@ class AppCenterViewController : NSViewController {
   @IBAction func userIdChanged(sender: NSTextField) {
     let text = sender.stringValue
     let userId = !text.isEmpty ? text : nil
-    UserDefaults.standard.set(userId, forKey: "userId")
+    UserDefaults.standard.set(userId, forKey: kMSUserIdKey)
     appCenter.setUserId(userId)
   }
+
+  // DeviceID
```
Comment (MatkovIvan) [inline SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift]:
```suggestion
  // Get device identifier.
```

File: SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift [hunk 9]
```diff
@@ -27,7 +34,23 @@ class AppCenterViewController : NSViewController {
...
+  class func getDeviceIdentifier() -> String? {
+    let platformExpert = IOServiceGetMatchingService(kIOMasterPortDefault, IOServiceMatching("IOPlatformExpertDevice"))
+    let serialNumberAsCFString = IORegistryEntryCreateCFProperty(platformExpert, kIOPlatformSerialNumberKey as CFString, kCFAllocatorDefault, 0)
+    let baseIdentifier = serialNumberAsCFString?.takeRetainedValue() as! String
+        IOObjectRelease(platformExpert)
+    return baseIdentifier
+  }
+
+  // Startup Mode
```
Comment (MatkovIvan) [inline SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift]:
```suggestion
  // Startup Mode.
```

**Pull request merged**